Standard.AI.Data.EntityIntelligence/Services/Foundations/Datas/DataService.cs

```diff
@@ -0,0 +1,69 @@
+﻿// ---------------------------------------------------------------------------------- 
+// Copyright (c) The Standard Organization, a coalition of the Good-Hearted Engineers 
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Standard.AI.Data.EntityIntelligence.Brokers.Datas;
+using Standard.AI.Data.EntityIntelligence.Models.Datas.Brokers;
+using Standard.AI.Data.EntityIntelligence.Models.Datas.Services;
+
+namespace Standard.AI.Data.EntityIntelligence.Services.Foundations.Datas
+{
+    internal partial class DataService : IDataService
+    {
+        private readonly IDataBroker dataBroker;
+
+        public DataService(IDataBroker dataBroker) =>
+            this.dataBroker = dataBroker;
+
+        public ValueTask<List<TableMetadata>> RetrieveTablesDetailsAsync() =>
+        TryCatch(async () =>
+        {
+            IEnumerable<TableColumnMetadata> retrievedTablesColumnsMetadata =
+              await this.dataBroker.ExecuteQueryAsync<TableColumnMetadata>(SelectAllTablesMetadataQuery);
+
+            return ToTablesMetadata(retrievedTablesColumnsMetadata);
+        });
+
+        private static List<TableMetadata> ToTablesMetadata(
+            IEnumerable<TableColumnMetadata> retrievedTablesColumnsMetadata)
+        {
+            var groupedColumns =
```

Comment:
You must explicitly declare the type of the left side if the right side isn't clear.
https://github.com/hassanhabib/CSharpCodingStandard/blob/master/5.%20Variables.md#do-6